src/array_api_typing/_array.py

```diff
@@ -0,0 +1,88 @@
+__all__ = (
+    "Array",
+    "BoolArray",
+    "HasArrayNamespace",
+    "NumericArray",
+)
+
+from pathlib import Path
+from types import ModuleType
+from typing import Literal, Never, Protocol, TypeAlias
+from typing_extensions import TypeVar
+
+import optype as op
+
+from ._utils import docstring_setter
+
+# Load docstrings from TOML file
+try:
+    import tomllib
+except ImportError:
+    import tomli as tomllib  # type: ignore[import-not-found, no-redef]
+
+_docstrings_path = Path(__file__).parent / "_array_docstrings.toml"
+with _docstrings_path.open("rb") as f:
+    _array_docstrings = tomllib.load(f)["docstrings"]
```

Comment:
It's a bit of a mouthful, but it helps with namespace pollution (I'm talking about the `f`).

```suggestion
_array_docstrings = tomllib.loads(_docstrings_path.read_text())["docstrings"]
```

And `_array_docstrings` will now be inferred as `Any`, so it could use an annotation (e.g. one wrapped as `Final`).

Also, since these are constants (the path too), uppercase names seem appropriate.